### ISSUE TITLE
Accessibility: No matching variations message not announced to screen reader users

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43613]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

On the product single, if selecting a variation that does not have options, a message displays on the screen that says, "Sorry, no products matched your selection. Please choose a different combination."

This message is not announced to screen reader users, as shown in this video:
https://github.com/woocommerce/woocommerce/assets/13153456/bd62a709-8d0b-4a99-adec-4187baa4cf8f

As a result, screen reader users will not know that their selection did not work until they explore the page.

### Expected behavior

The wc-no-matching-variations message needs to use [aria-live](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-live) to ensure that it is announced by screen readers.

This is the section of code where it needs to be added:
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce/client/legacy/js/frontend/add-to-cart-variation.js#L191

### Actual behavior

See video linked above.

### Steps to reproduce

1. Turn on a screen reader such as VoiceOver or NVDA.
2. Go to a variable product and select options for which you know there is not a variation available.
3. Observe that the status message appears on the screen but that the screen reader does not announce the message.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by dinhtungdu** [comment]
I added this issue to our backlog. Thank you so much for reporting this.

**Comment by Manussakis** [comment]
@amberhinds Thank you for reporting this issue!
I couldn't reproduce it. Maybe something has changed in the code base since this issue was created.

It seems to me that only combinations of variations that exist are available to be selected by users.
For example, on my setup, blue Hoodies don't have the medium size:
![image](https://github.com/user-attachments/assets/f63b0b45-99b3-4725-819c-1dc19b05a4b3)

Once I select the blue color, the medium size is hidden from the size input.
![image](https://github.com/user-attachments/assets/b29b63f8-b983-46bd-aa9f-ed0946419083)

Could you please confirm if this issue still happens on your side and give me more context on how to reproduce it if it does.

**Comment by amberhinds** [comment]
I still get this issue on this product: https://shop.equalizedigital.com/product/make-wordpress-accessible-dark-mode-t-shirt/

1. Set color to Forest
2. Size dropdown does not automatically change to only show available sizes for the Forest color.
<img width="504" alt="Size dropdown expanded showing all possible sizes" src="https://github.com/user-attachments/assets/8a385ec6-a795-4d57-9486-18e32c1b9243">

3. Choose 5x as the size and the message will appear.
<img width="876" alt="No products available message" src="https://github.com/user-attachments/assets/3cad819d-f0cd-4717-ac22-40b0d509469e">

Here are the variations for this shirt with Forest as the color:
<img width="1038" alt="Variation screenshot - only goes up to 4XL" src="https://github.com/user-attachments/assets/fc49fad2-6307-4c91-9a50-83a54b7bb643">

This product single is built with FSE and using the Add to Cart with Options block.

**Comment by Manussakis** [comment]
@amberhinds I was able to reproduce the issue. The AJAX request for variations only happens if the number of variations is higher than the number returned by the `woocommerce_ajax_variation_threshold` filter, which is 30 by default.
If the number is lower, the validation happens dynamically on the client side by enabling/disabling options based on the previous option selection.
